wgpu-hal/src/metal/device.rs

```diff
@@ -219,6 +219,13 @@ impl super::Device {
         }
     }
 
+    pub unsafe fn device_from_raw(raw: mtl::Device, features: wgt::Features) -> super::Device {
```

Comment:
Are there any safety invariants that need to be upheld by these functions? These kind of functions need their invariant well documented.

Reply:
I am not sure, I'm not familiar with Metal enough. None of the other functions in there have any safety comments though